pkg/controller/elasticsearch/remotecluster/elasticsearch.go

```diff
@@ -0,0 +1,129 @@
+	if err != nil {
+		return err
+	}
+	if !enabled {
+		log.Info(
+			enterpriseFeaturesDisabledMsg,
+			"namespace", es.Namespace, "es_name", es.Name,
+		)
+		eventRecorder.Eventf(&es, corev1.EventTypeWarning, events.EventAssociationError, enterpriseFeaturesDisabledMsg)
+		return nil
+	}
+
+	currentRemoteClusters, err := getCurrentRemoteClusters(es)
+	if err != nil {
+		return err
+	}
+	if currentRemoteClusters == nil {
+		currentRemoteClusters = make(map[string]string)
+	}
+	expectedRemoteClusters := getExpectedRemoteClusters(es)
+
+	remoteClusters := make(map[string]esclient.RemoteCluster)
+	// RemoteClusters to add or update
+	for name, remoteCluster := range expectedRemoteClusters {
+		if currentConfigHash, ok := currentRemoteClusters[name]; !ok || currentConfigHash != remoteCluster.ConfigHash {
+			// Declare remote cluster in ES
+			seedHosts := []string{services.ExternalTransportServiceHost(remoteCluster.ElasticsearchRef.NamespacedName())}
+			log.Info("Adding or updating remote cluster",
+				"namespace", es.Namespace,
+				"es_name", es.Name,
+				"remote_cluster", remoteCluster.Name,
+				"seeds", seedHosts,
+			)
+			remoteClusters[name] = esclient.RemoteCluster{Seeds: seedHosts}
+		}
+	}
+
+	// RemoteClusters to remove
+	for name := range currentRemoteClusters {
+		if _, ok := expectedRemoteClusters[name]; !ok {
+			log.Info("Removing remote cluster",
+				"namespace", es.Namespace,
+				"es_name", es.Name,
+				"remote_cluster", name,
+			)
+			remoteClusters[name] = esclient.RemoteCluster{Seeds: nil}
+			delete(currentRemoteClusters, name)
+		}
+	}
+
+	if len(remoteClusters) > 0 {
+		// Apply the settings
+		if err := updateSettings(esClient, remoteClusters); err != nil {
+			return err
+		}
+		// Update the annotation
+		return annotateWithRemoteClusters(c, es, expectedRemoteClusters)
+	}
+	return nil
+}
+
+// getExpectedRemoteClusters returns a map with the expected remote clusters
+// A map is returned here because it will be used to quickly compare with the ones that are new or missing.
+func getExpectedRemoteClusters(es esv1.Elasticsearch) map[string]expectedRemoteClusterConfiguration {
+	remoteClusters := make(map[string]expectedRemoteClusterConfiguration)
+	for _, remoteCluster := range es.Spec.RemoteClusters {
+		if !remoteCluster.ElasticsearchRef.IsDefined() {
+			continue
+		}
+		remoteCluster.ElasticsearchRef = remoteCluster.ElasticsearchRef.WithDefaultNamespace(es.Namespace)
+		remoteClusters[remoteCluster.Name] = expectedRemoteClusterConfiguration{
+			RemoteCluster: remoteCluster,
+			ConfigHash:    remoteCluster.ConfigHash(),
+		}
+	}
+	return remoteClusters
+}
+
+// updateSettings makes a call to an Elasticsearch cluster to apply a persistent setting.
+func updateSettings(esClient esclient.Client, remoteClusters map[string]esclient.RemoteCluster) error {
+	ctx, cancel := context.WithTimeout(context.Background(), esclient.DefaultReqTimeout)
+	defer cancel()
+	return esClient.UpdateSettings(ctx, esclient.Settings{
```

Comment:
I didn't know that. It does not seem dynamic, a cluster restart would be needed ?

Reply:
Yes, I think we would restart the nodes every time a user would change the remote cluster configuration. So it's a trade-off between avoiding blocking API calls from the operator and potential disruption of production workloads managed by ECK because we restart nodes too often. 

So +1 on keeping it the way it is now.